### PR TITLE
fix (flake): use same node version for corepack

### PR DESCRIPTION
**Pull request by sectore**
## Type

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [x] Optimization
- [ ] Documentation Update

## Description

- `flake.nix`: 
   - Use `corepack` of `nixpkgs
   - Remove `flake-utils`
   - Remove uneeded `bun` + `nodePackages.json` from `buildInputs`
- `flake.lock`: Update by running `nix flake update`

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -17,9 +17,10 @@
   }:
     flake-utils.lib.eachDefaultSystem (system: let
       pkgs = nixpkgs.legacyPackages.${system};
+      node = pkgs.nodejs_20;
       corepackEnable = pkgs.runCommand "corepack-enable" {} ''
```
Comment (IMax153) [inline flake.nix:21]: As it turns out the `corepack` package in `nixpkgs` already enables itself if included in the derivation, so if you're up for it @sectore you can just copy what we did in the `flake.nix` in the [`effect` repository](https://github.com/Effect-TS/effect/blob/main/flake.nix).

**Review by IMax153 — CHANGES_REQUESTED** [review]

File: flake.nix (hunk 1 above)
Comment (sectore) [reply flake.nix]: Done 382024b . 

Not sure if `bun` is still needed here https://github.com/Effect-TS/website/blob/382024b31e0b1adea198fd8e1833b43c65d2bc7c/flake.nix#L18 There is a [pnpm-lock.yaml](https://github.com/Effect-TS/website/blob/main/pnpm-lock.yaml) in the repo only.  (just let me know)

Comment (IMax153) [reply flake.nix]: I believe both bun and nodePackages.json can be removed.

Comment (sectore) [reply flake.nix]: Done 9e7068d

**Comment by IMax153** [comment]
@sectore - apologies that this got lost. Thank you for the PR 🙏 - will merge once CI builds.

**Review by IMax153 — APPROVED** [review]

**Pull request merged**